plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -2057,24 +2058,31 @@ public void testSplitPruningForFilterOnPartitionColumn()
         // sanity check that table contains exactly 5 files
         assertThat(query("SELECT count(*) FROM \"" + tableName + "$files\"")).matches("VALUES CAST(5 AS BIGINT)");
 
-        verifySplitCount("SELECT * FROM " + tableName, 5);
-        verifySplitCount("SELECT * FROM " + tableName + " WHERE regionkey = 3", 1);
-        verifySplitCount("SELECT * FROM " + tableName + " WHERE regionkey < 2", 2);
-        verifySplitCount("SELECT * FROM " + tableName + " WHERE regionkey < 0", 0);
-        verifySplitCount("SELECT * FROM " + tableName + " WHERE regionkey > 1 AND regionkey < 4", 2);
+        verifySplitCount("SELECT * FROM " + tableName, 5, computeActual("SELECT * FROM nation"));
+        verifySplitCount("SELECT * FROM " + tableName + " WHERE regionkey = 3", 1, computeActual("SELECT * FROM nation WHERE regionkey = 3"));
```

Comment:
nit: you can move `computeActual` to `verifySplitCount` to have shorter call sites.
```
verifySplitCount(String query, int expectedSplitCount, String expectedValuesQuery) {
    expectedValues = computeActual(expectedValuesQuery);
    verifySplitCount(query, expectedSplitCount, expectedValues);
}


verifySplitCount(String query, int expectedSplitCount, MaterializedResult expectedValues) {


}

Reply:
i would go with `verifySplitCount(String query, int expectedSplitCount)` signature (original one)

inside the method, you can run `query` disabling all pushdowns to verify correctness
`io.trino.sql.query.QueryAssertions.QueryAssert#verifyResultsWithPushdownDisabled`